dj/models/node.py

```diff
@@ -722,6 +725,61 @@ class SparkConf(BaseSQLModel):
     __root__: Dict[str, str]
 
 
+class PartitionType(str, enum.Enum):
+    """
+    Partition type.
+
+    A partition can be temporal or categorical
+    """
+
+    TEMPORAL = "temporal"
+    CATEGORICAL = "categorical"
+
+
+class Partition(BaseSQLModel):
+    """
+    A partition specification tells the ongoing and backfill materialization jobs how to partition
+    the materialized dataset and which partition values (a list or range of values) to operate on.
+    Partitions may be temporal or categorical and will be handled differently depending on the type.
+
+    For temporal partition types, the ongoing materialization job will continue to operate on the
+    latest partitions and the partition values specified by `values` and `range` are only relevant
+    to the backfill job.
+
+    Examples:
+        This will tell DJ to backfill for all values of the dateint partition:
+          Partition(name=“dateint”, type="temporal", values=[], range=())
+        This will tell DJ to backfill just 20230601 and 20230605:
+          Partition(name=“dateint”, type="temporal", values=[20230601, 20230605], range=())
+        This will tell DJ to backfill 20230601 and between 20220101 and 20230101:
+          Partition(name=“dateint”, type="temporal", values=[20230601], range=(20220101, 20230101))
+
+        For categorical partition types, the ongoing materialization job will *only* operate on the
+        specified partition values in `values` and `range`:
+            Partition(name=“group_id”, type="categorical", values=["a", "b", "c"], range=())
+    """
+
+    name: str
+    values: Optional[List]
+    range: Optional[Tuple]
```

Comment:
`values` and `range` make sense... and perhaps for the single value that we gonna need for the temporal partition in **scheduled** runs we could add an `expression: Optional[str]` here?

Reply:
Yep, that makes sense! I'll add that